Feature/kas 2667 new agenda overview

sergiofenoll opened the pull request:
Draft for tests, I'm guessing many will fail because of changed routes/missing `data-test` attributes

sergiofenoll committed:
Remove agendas.overview

sergiofenoll committed:
Re-add filter & use loading state

sergiofenoll committed:
Add size to queryParams of agendas route

sergiofenoll committed:
Re-enable reset filter button

sergiofenoll committed:
Update failing tests

sergiofenoll committed:
No longer expect /overzicht to exist in tests

sergiofenoll committed:
Merge branch 'development' into feature/KAS-2667-new-agenda-overview

sergiofenoll committed:
Merge branch 'development' into feature/KAS-2667-new-agenda-overview

sergiofenoll committed:
Fix selector in test & passing data-test attribute & anchor

sergiofenoll committed:
Always sort agendas on status

erikap commented on app/pods/agendas/route.js:
```diff
@@ -1,17 +1,73 @@
+    transition.promise.finally(() => {
+      controller.set('isLoadingModel', false);
+    });
+
+    // only bubble loading event when transitioning between tabs
+    // to enable loading template to be shown
+    if (transition.from && transition.to) {
+      return transition.from.name != transition.to.name;
+    } else {
+      return true;
+    }
```
Dit lijkt mij hier overbodig?  We willen de loading state nooit bubblen aangezien we die zelf beheren met `isLoadingModel` en deze route verder geen subroutes heeft.

sergiofenoll replied:
Deze heb ik om eerlijk te zijn gewoon ge-copy-paste van een andere route waar dezelfde constructie toegepast wordt. :sweat_smile: Maar is hier indd niet nodig.

erikap commented on app/components/agenda/agenda-header/agenda-tabs.hbs:
```diff
@@ -1,11 +1,7 @@
+  <Auk::Tab @route="agendas" @isHierarchicalBack={{true}} />
   <Auk::Tab
     @route="agenda.agendaitems.index"
-    @query={{if
-      this.currentAgendaItemId
-      (hash anchor=this.currentAgendaItemId)
-    }}
```
Waarom is deze query param verwijderd? Die zorgt ervoor dat in het detail automatisch naar het juiste agendapunt gescrolled wordt afaik.

sergiofenoll replied:
De `should test scroll anchor` test van `agendaitem-changes.spec.js` faalde op [deze lijn](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/cypress/integration/unit/agendaitem-changes.spec.js#L274). Probleem zijnde dat als je in de "Overzicht" bent, dan op de "Details" klikt, dan terug "Overzicht" de originele anchor verandert, waardoor er niet gescrolld werd naar het juiste agendapunt.

Hoe ik de error begreep:

Op [deze lijn in `agenda-tabs.js`](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/components/agenda/agenda-header/agenda-tabs.js#L33) krijgen we de modellen die aan de route van de "Details" tab gegeven worden. Vanuit "Overzicht" gaat `currentAgendaItemId` null zijn, want we zitten daar in de `agenda.agendaitems.index` route, niet in een subroute van `agenda.agendaitems.agendaitem`. Dus de route voor de "Details" tab zal naar het agendapunt `firstAgendaitem` gaan. Eens daar wordt de route voor de "Overzicht" opnieuw berekend, en deze keer is er wel een `currentAgendaItemId`, nl. de `firstAgendaitem`. Dus de anchor wordt overschreven met een foute/onverwachte waarde.

Na nader inzien is dit wel geen echte fix, want als er van agendapunt geswitcht wordt in de "Details" view, dan wordt die anchor niet juist gezet

erikap commented on app/pods/agendas/controller.js:
```diff
@@ -7,9 +7,54 @@ import { action } from '@ember/object';
+    if (!match) {
+      this.from = null;
+      this.to = null;
+      return;
```
Persoonlijk vind ik het hier duidelijker/logischer om met if-else te werken ipv met early-return.
 
Als we `this.from` en `this.to` aanpassen (en dus model refresh triggeren), kunnen we `this.page` best ook naar 0 resetten.

erikap reviewed (changes requested):
Styling ook nog over te nemen vanop usertest branch

sergiofenoll committed:
No need to bubble loading state

sergiofenoll committed:
Apply styling & logic changes from usertest branch

sergiofenoll committed:
Make entire row clickable again

sergiofenoll committed:
Add data-test selector for filter input to agendas

sergiofenoll committed:
Correct condition for showing ChangesAlert

sergiofenoll committed:
Check on nullish instead of empty string for this.filter

sergiofenoll committed:
Merge branch 'development' into feature/KAS-2667-new-agenda-overview

sergiofenoll committed:
Fix behaviour when deleting agendaitems and navigating between agenda tabs

sergiofenoll commented on app/components/agenda/agenda-header/agenda-tabs.js:
```diff
@@ -11,27 +10,12 @@ export default class AgendaAgendaHeaderAgendaTabsComponent extends Component {
-      });
-    }
-    return null;
+  get firstAgendaitem() {
```
Some explanation on this change.

When deleting the last agendaitem on an agenda, the "Details" tab is still there. But clicking it breaks the app, because it's trying to navigate to an agendaitem that no longer exists. I couldn't find a way to force the `AgendaTabs` component to refire this task, so instead I passed along the agendaitems to this component, from the route's model. So they get fetched from the backend by the `model()` hook of `agenda`'s route and passed down the component chain, and when we delete an agendaitem, since the model gets recalculated the component will get rerendered.

sergiofenoll commented on app/components/agenda/agenda-header/agenda-tabs.js:
```diff
@@ -41,11 +25,16 @@ export default class AgendaAgendaHeaderAgendaTabsComponent extends Component {
+      while (agendaItemsRoute.name !== 'agenda.agendaitems') {
+        agendaItemsRoute = agendaItemsRoute.parent;
+      }
+      return agendaItemsRoute.queryParams.anchor;
```
When going from "Overview" to "Details" and back to "Overview" via the tabs, the anchor would be reset to the first agendaitem, regardless of whether a different agendaitem was previously selected. This occurred because in the "Overview" tab we are in the `agenda.agendaitems` route. In the original code, we could never get an agendaitem's id, so the route of "Details" tab defaulted to the first agendaitem, and then coming back to "Overview" the anchor would get replaced with the id of the first agendaitem.

To fix this, we try to get the agendaitem id from the anchor query parameter if we failed to get it from the route itself.

sergiofenoll committed:
Allow immediate setting of filter by pressing enter

This is potentially useful for the user, but also especially for the
Cypress tests so we don't need to wait for the debounce.

Also tell Cypress to explicitly wait for the loading state of the table
to be gone. Because now use an inline loading state as opposed to
ember's route based loading system, Cypress was trying to click on the
table rows too quickly, before the actual results had rendered, which
caused the tests to fail.

sergiofenoll committed:
Tweak default sorting of agendas table & QP error

The session-kind test expects the PVV Ministerraad to come after the
normal Ministerraad in a filtered agendas table. With this new
default filtering, we explicitly apply this ordering.

An agendaitem delete call was causing errors which might be related to
these Ember bugs:

- https://github.com/emberjs/ember.js/issues/20051
- https://github.com/emberjs/ember.js/pull/19971

Leaving out setting anchor to null in hope that it might fix it.

sergiofenoll committed:
Pass meeting & agenda to AgendaitemNav component

sergiofenoll committed:
Use optional chaining for agendaitem in nav

ValenberghsSven commented on app/pods/agendas/controller.js:
```diff
@@ -2,15 +2,45 @@ import Controller from '@ember/controller';
+  @tracked filter = null;
+  @tracked page = 0;
+  @tracked size = 10;
+  @tracked sort = '-status,created-for.planned-start,created-for.kind.label';
```
The most recent agenda is on the bottom of the list? 
guessing you need `-created-for.planned-start`

sergiofenoll replied:
Ah, didn't realize we wanted it the other way around. Will change.

ValenberghsSven commented on app/components/auk/pagination.hbs:
```diff
@@ -1,7 +1,7 @@
 <div class="auk-pagination">
   <ul class="auk-pagination__list">
     <li class="auk-pagination__element">
-      <p><strong>{{this.startItem}}-{{this.endItem}}</strong> {{t "of"}} {{this.totalNbOfItems}}</p>
```
This changes every view that uses pagination, is this desired?
Assuming so, just checking.

sergiofenoll replied:
Not sure, the request was to follow the styling of the usertest branch, which didn't have bolded numbers in the footer. Thought it'd make more sense to change it everywhere instead of only here.

erikap replied:
Changing everywhere seems desired to me :+1:

erikap replied:
Sorry, I interpreted this change wrong. The `<strong>` tag should **not** be removed. Usertest branch still uses the `NumberPagination`-component (embedded in the `DataTable`) component. That's why the page numbers are not bold.
But the new `Auk::Pagination` component - which we are gradually transition to in the entire app - has bold page numbers intentionally. 

TLDR; We should keep the `<strong>` tag. It's OK that this is not in line with the usertest branch.

ValenberghsSven commented on app/router.js:
```diff
@@ -17,9 +17,7 @@ Router.map(function() {
-  this.route('agendas', { path: '/', }, function() {
-    this.route('overview', { path: '/overzicht', });
-  });
+  this.route('agendas', { path: '/', });
```
`path: '/overzicht'` ?
The adress for agendas is now just the homepage of the app.
It used to be like that before, and with redirects we ended up on `/overzicht`, but not having an address `/overzicht` looks a little strange?
As seen on the usertest, the address is still  `/overzicht`.


![image](https://user-images.githubusercontent.com/22245223/164009241-2391f662-135f-427a-b1c6-a2caadf4cd3a.png)

![image](https://user-images.githubusercontent.com/22245223/164009309-96f47d14-afb7-4e3f-8523-c676292eafe4.png)

sergiofenoll replied:
I took this from the ticket
> Aangezien dit impact heeft op de resultaten die getoond worden in de tabel en er geen andere subroutes zijn, kunnen de huidige routes agendas en agendas.overview best samengevoegd worden tot 1 route agendas.

to mean that we wanted this as the root page of the app. I might have misunderstood though

ValenberghsSven replied:
Not sure what we want, maybe we should check with design team and users.

erikap replied:
Reworked route structure is ok, but I would keep `path: '/overzicht'` as @ValenberghsSven suggests on the root `agendas` route such that it stays the same as it is now.

sergiofenoll replied:
Dumb question but how do I get the app to automatically redirect to `/overzicht`? If the `agendas` route is set to `/overzicht` a user that navigates to `/` appears in the application route (so the page is empty)? Do I have to add a `redirect` in that route?

EDIT: Added a redirect to route, but checking based on the `transtion.to.name` to only perform the redirect when going to `index`.

erikap replied:
@sergiofenoll The ember-way to do the redirect is creating a root `index/route.js` (without controller/template) and transition to `agendas.overview` in the `beforeModel`-hook of that route.

ValenberghsSven commented on app/pods/agendas/template.hbs:
```diff
@@ -19,12 +40,176 @@
+                >
+                  <td data-test-route-agendas-overview-row-title>
+                    <span class="auk-o-flex auk-o-flex--vertical-center auk-o-flex--spaced">
+                      {{#if agenda.status.isDesignAgenda}}
+                        <strong>{{t "design-agenda"}}</strong>
+                        <Auk::Pill @skin="active-future">{{agenda.name}}</Auk::Pill>
+                      {{else}}
+                        <strong>{{t "agenda"}}</strong>
+                        <Auk::Pill>{{agenda.name}}</Auk::Pill>
+                      {{/if}}
+                      <span>–</span>
+                      <span>
+                        {{moment-format agenda.createdFor.plannedStart "DD.MM.YYYY"}}
```
`status` and `createdFor` are not included in the model query and not loaded anywhere else in the route or controller.
So this data is not available when loading the template and we are technically "awaiting" the data here, even if we are not explicitly mentioning it.

You can see the data changing during loading:
- the dates of the agendas is wrong when loading the page. Every single date is showing today's date until the data is loaded and after that the date is correct.
- status is showing "agenda", and changes to "designagenda" later.
- kind is missing while loading.

sergiofenoll replied:
If we preload the data, on my machine, with only 43 agendas, loading the frontpage already takes ~5 seconds. I will try with more, maybe the time doesn't increase linearly with the amount of agendas, but I do believe 5 seconds is already too slow.

sergiofenoll replied:
tried it out with 1100 agendas, that goes up to 8 seconds. not as bad as I expected, but frankly anything above 1 second is not great.

imo we can tackle this
- by making mu-cl-resources and/or mu-authorization faster (maybe this isn't actually feasible though)
- by creating a microservice that fetches the agendas. maybe the handwritten SPARQL query is faster than mu-cl-resource?
- taking the hit of lazy loading, but using skeleton loaders (or a different visual mechanism) to make clear to the user that we're still loading data, but that they can interacting with a part of the site

ValenberghsSven replied:
I think this is a consequence of querying for agendas instead of meetings. There are just way more of them.
But this should improve with the addition of the filter on `has-no:next-version`?

ValenberghsSven replied:
Also, we should only load the 20 or so agendas that are shown with page sizes, what do you mean "tried it out with 1100 agendas"?

sergiofenoll replied:
testing was done using `'filter[:has-no:next-version]': true,` as part of the query params. i also just created 1100 separate agendas (as if the "create new agenda" & modal was filled out 1100 times), so it would've been the same if we were querying for meetings or agendas (same amount of each, though in practice we'll have more agendas than meetings)

sergiofenoll replied:
> Also, we should only load the 20 or so agendas that are shown with page sizes, what do you mean "tried it out with 1100 agendas"?

I created 1100 agenda's and then queryed normally (so with page size 10, as is the default). As there's more data in the database, querying gets slower.
Because we're filtering and sorting, the DB has to deal with the whole dataset.

ValenberghsSven replied:
"Because we're filtering and sorting, the DB has to deal with the whole dataset."
We are using pagination, so unless you are loading the page with `/?size=1100` it shouldn't take that long to load 10 agendas (default page size)?
The problem with the current code is that we are showing **wrong/missing data** between the initial load and when everything is finally loaded.
also a spam of seperate calls for each of that data is happening, which will also impact performance with larger page sizes and can put more stress on the backend.
![image](https://user-images.githubusercontent.com/22245223/164186849-acf246d9-efd7-420d-8cd5-7f6f24da246b.png)

ValenberghsSven replied:
Still loading, date is wrong on some agendas (today's date), kind is missing
![still loading](https://user-images.githubusercontent.com/22245223/164187884-52e17b4d-426e-4fb7-9271-b796d83cc9b9.png)

Done loading
![loaded](https://user-images.githubusercontent.com/22245223/164187877-16c53229-19a4-4ead-8f8b-a91fe5a55eaf.png)

sergiofenoll replied:
Yes, we're doing pagination, but even if we're only receiving 10 agendas, the DB still needs to go over all the agendas it has to know which 10 agendas to give us. It needs to remove all agendas which do not match our filtering criteria from the result set, and then it needs to go over all those agendas to sort them according to the sort options we requested.

If we were only requesting 10 agendas, no sorting or filtering involved, then yes this would be fast and the total amount of agendas in the backend wouldn't matter. But because we're filtering and sorting on top of the pagination, as the amount of data grows, the query will get slower.

I agree that the extra calls aren't ideal either. But it's a trade-off between 1 call that is slow (in our case, very slow), or a number of calls to lazy-load data.

I also agree that showing wrong data while it is loading is bad. We could use [skeleton loading](https://betterprogramming.pub/the-what-why-and-how-of-using-a-skeleton-loading-screen-e68809d7f702) for that, showing no data, and instead a grey square. This way the user knows data is still loading while still seeing something.

ValenberghsSven replied:
It's ok if the loading of kind happens after the data is shown, that is also what happens in the current implementation.
My main problem is that we are showing wrong dates and that we are showing "agenda" before we fully know the status so it changes to "ontwerpagenda" later.

My main point is that we should show no data instead of "seemingly" wrong data.

ValenberghsSven replied:
" But it's a trade-off between 1 call that is slow (in our case, very slow), or a number of calls to lazy-load data."

The agendas query in question is not that complex, currently there are less than 3000 agendas on prod DB.
Even with filtering and sorting, you would expect a fast query.
Either we are querying wrong or there is a problem in the backend.

erikap replied:
@sergiofenoll When adding > 1000 agenda's in the triplestore, are you running Virtuoso with the default config or with a config tuned for that amount of data [like on production](https://github.com/kanselarij-vlaanderen/app-kaleidos/blob/development/config/db/virtuoso.production.ini)? Without that tuned config the timings you put here are not representative.

When I run this branch with a copy of the ACC DB (incl. config), containing > 2000 agenda's, the query as it is now takes 2s on my machine with page size 25 and a cold cache and cold mu-cl-resource. Additional data is loaded in parallel in the magnitude of 1-2s. I don't see wrong data appear before the right data.

It is known that include is a very expensive operation in mu-cl-resources (there are plans to improve this in the future).  For smaller page sizes I don't see a problem in the additional requests. Hannes' opinion about the page size options (which I agree on) is that we should not offer page size options that we now will be too slow to be used in practice. So, regardless we go for include or not, I would remove page size options > 100.

sergiofenoll replied:
didn't realize there was a different config for production, also didn't realize it'd made such a big difference. i tried it using the dev configuration.

for what it's worth, the numbers i shared were also with a cold cache and cold mu-cl-resources, but *with* `include: 'status,created-for,created-for.kind'`, which is not what's currently committed.

sergiofenoll replied:
Right now `include: 'status,created-for,created-for.kind'` is committed. The query fetching the agendas is somewhat slow because of this, but we get the expected data from the first load.

What's the preferred solution in this case, live with the somewhat  slow query, or lazy load the agenda's meeting?

ValenberghsSven replied:
Go with it for now? We can evaluate on ACC with larger agendas on a faster server.

ValenberghsSven commented on app/pods/agendas/route.js:
```diff
@@ -3,15 +3,74 @@ import { inject as service } from '@ember/service';
+        number: params.page,
+        size: params.size,
+      },
+    };
```
See previous comment, do we want to include `status,created-for` here to avoid data loading in the template?
Might impact performance with lots of agendas

ValenberghsSven replied:
the included is in comment and being discussed, closing this conversation

ValenberghsSven commented on app/pods/agendas/route.js:
```diff
@@ -3,15 +3,74 @@ import { inject as service } from '@ember/service';
+    return await this.store.query('agenda', queryParams);
+  }
+
+  @action
+  loading(transition) {
+    // see snippet in https://api.emberjs.com/ember/3.27/classes/Route/events/loading?anchor=loading
+    // eslint-disable-next-line ember/no-controller-access-in-routes
+    const controller = this.controllerFor(this.routeName);
+    controller.set('isLoadingModel', true);
+    transition.promise.finally(() => {
+      controller.set('isLoadingModel', false);
+    });
```
This codeblock has been used more and more throughout the app.
Out of scope, could we turn it this into a util or something to avoid code duplication?
Could be placed on backlog for later

ValenberghsSven commented on cypress/integration/all-flaky-tests/mandatee-assigning.spec.js:
```diff
@@ -334,18 +334,30 @@ context('Assigning a mandatee to agendaitem or subcase should update linked subc
-      .contains('Op voorstel van minister-president Jan Jambon en viceminister-president Hilde Crevits');
+      .scrollIntoView()
+      .contains('Op voorstel van minister-president Jan Jambon en viceminister-president Hilde Crevits')
+      .should('be.visible');
```
testing `should('be.visible')` seems futile after using `scrollIntoView()`.
A `contains` shouldn't fail when the selected element is not in viewport, reasoning behind the scrolling?

sergiofenoll replied:
This test was failing for me because (from what I could tell), the item with the `Op voorstel van ...` string was not visible, which is why I added the `scrollIntoView` & `should('be.visible')`. Although the two probably aren't needed together.

ValenberghsSven commented on app/components/agenda/agendaitem/agendaitem-nav.js:
```diff
@@ -23,23 +23,23 @@ export default class AgendaitemNav extends Component {
       }
       if (this.decisionsExist) { // Treatment and decision activity are currently one entity in implementation
-        const treatment = (await this.agendaitem.get('treatments')).firstObject;
+        const treatment = (await this.args.agendaitem?.treatments)?.firstObject;
```
We already checked that there is an agendaitem and treatments with `this.decisionsExist = isPresent(await this.args.agendaitem?.treatments);`
Do we still need optional operators for this check when `this.decisionsExist` is true?

ValenberghsSven commented on app/pods/agenda/route.js:
```diff
@@ -25,10 +25,15 @@ export default class AgendaRoute extends Route {
       sort: '-serialnumber',
       include: 'status',
     });
+    const agendaitems = await this.store.query('agendaitem', {
+      'filter[agenda][:id:]': agenda.id,
+      sort: 'show-as-remark,number',
+    });
```
Currently, the only reason for this is the getter `firstAgendaitem` in `agenda/agendaitems`.
The route `agenda/agendaitems` has its own query for agendaitems and is not the same.

https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-2667-new-agenda-overview/app/pods/agenda/agendaitems/route.js#L33

Loading agendaitems again here will slow down the overall process of loading the agenda, impacting performance.
This query is not cached by `cache-warmup-service`, unlike the one in the other route.
Also I wouldn't recommend loading agendaitems in the top route, since you can visit `/documents` or `/compare` routes and you don't need the agendaitems loaded in that case. It makes more sense to keep them in the route `agenda/agendaitems`.

Either queryOne agendaitem as the `firstAgendaitem` and pass that down, or just load it with a task where you need it in `AgendaTabs` since that is the only place.

sergiofenoll replied:
I load the agendaitems here in the agenda route because we do need them, for rendering the `agenda-tabs` in the header. These tabs are rendered anytime an agenda is opened, even in the `/documents` and `/compare` subroutes. The reason for passing down the agendaitems to the component is to force it ro re-render when the agendaitems change. E.g. when you delete the last agendaitem in an agenda, the "Details" tab still exists because the task to load the agendaitems does not refire. Then clicking on that tab breaks the app. See: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1288/files/e80d2828b48d5cdd442a1d937d23051383f008bd#r851374926

Didn't realize it got loaded in again down the line, but the query in this route can be made conform with the `agenda.agendaitems` query, and in the `agenda.agendaitems` route we get it from the `agenda` model.

ValenberghsSven commented on app/pods/agendas/controller.js:
```diff
@@ -26,6 +56,35 @@ export default class AgendasController extends Controller.extend(DefaultQueryPar
+
+  @action
+  async onClickRow(agenda) {
+    const meeting = await agenda.createdFor;
```
check other comment, `createdFor` should already be loaded for other reasons.

sergiofenoll replied:
Left in the async & await call here because even if we include `createdFor`, `agenda.createdFor` returns a proxy and we'd have to either `await` it or use `meeting.get()`.

ValenberghsSven commented on app/styles/auk/_auk-additions.scss:
```diff
@@ -160,3 +160,104 @@ table tr.lt-expanded-row {
     fill: currentColor;
   }
 }
+
+$vl-picton-blue: #16465B;
+$vl-gray-70: #687483;
+$vl-gray-50: #8695A8;
```
Css additions, is this temporary so we can change all the views and then switch the css from the package?

Also, you only define 3 colors but end up using a bunch more hex code colors in this file.

sergiofenoll replied:
Honestly just copy-pasted these from the usertest branch. Haven't looked at what is strictly necessary and what can be cleaned up yet.

brenner-company replied:
I think the cleanup of this will be something for me. I'll have to go through all the additions (these and previous ones) and tweak/move (to `au-kaleidos-css`) styling where needed. Blocking this PR won't be needed because I'm only be able to dedicate time to this in the following spint (85).

ValenberghsSven commented on app/pods/agendas/route.js:
```diff
@@ -3,15 +3,74 @@ import { inject as service } from '@ember/service';
+      }
+    }
+
+    return await this.store.query('agenda', queryParams);
```
You are querying all agendas resulting in meetings with multiple agendas to be in your set multiple times.
You probably want to filter on the highest Version by filtering on `has-no:nextVersion` or something.
Or go back to queries on `meeting`, but that's probably not desired.

![image](https://user-images.githubusercontent.com/22245223/164028350-ffc42d41-fa00-4b24-85c0-1750a277c780.png)

ValenberghsSven reviewed (changes requested):
`Agenda::FutureAgendas` still exists, cleanup
Also cleanup `latestAgendaName` computed on meeting, only used in `FutureAgendas`


Still looking at the tests, not sure why some of the profile tests are failing

sergiofenoll committed:
Fix default sorting & filter of agendas overview

ValenberghsSven commented:
Looking at other PR's which still use the current implementation i've noticed more things.
The current one has 2 lists:
- the active meetings (planned start is after today and meeting is not closed yet)
- history (all meetings).

They each have their own sorting on plannedStart (one Asc, other Desc).

The new one can only have 1 sorting, and I was assuming that we wanted sorting on `status` and then descending on `planned-start`. The notion file confirms this  "Agenda’s Sorteren op status Geopend > Gesloten. De volgorde hier is van recent naar oud".
But what are we sorting on when we do `sort= '-status'`? Are we sorting on the uri? There are 3 uri's. design/approved/closed.
Most of the agendas will either be `design` or `closed`.
If you delete the design agenda of a meeting that has other approved agendas, you will end up with an agenda with status `approved`. Where is that one sorted? It could be on top, in the middle or on the bottom but you expect it to be on the same level as `design`. This can happen/has happened on PROD. That agenda might "disappear" from the overview because of the sorting, depending on where it ends up.

I think what you want is to sort on`created-for.is-final`

erikap commented:
> I think what you want is to sort on`created-for.is-final`

I think you're right @ValenberghsSven. I wasn't aware of the use case where the latest design agenda is deleted and the latest agenda status differs from the `is-final` flag on meeting. The `is-final` flag on meeting is then probably also what we want to check in the data-table in the status column `{{#if meeting.isFinal}}Afgesloten{{else}}Geopend{{/if}}` (sorry for the confusion in the story @sergiofenoll)

sergiofenoll committed:
Undo extraneous changes unrelated to this PR

sergiofenoll committed:
Undo unwanted changes

agendas is /overzicht again & bolded numbers in table pagination

sergiofenoll committed:
Load meeting of agenda up front

sergiofenoll commented:
Ter info, nu ziet de lijst er (mogelijks) zo uit
![image](https://user-images.githubusercontent.com/22411874/164228126-a3ecd443-5c66-4e09-8ed5-8f42fef80cde.png)
Dit is gegeven dat er een vergadering niet afgesloten is waarvan de ontwerpagenda verwijderd is. Dan komt in de lijst enkel de "finale" agenda terecht (dus zonder stippelijn) met ernaast als status "Geopend".

ik vermeld het omdat in mijn ogen de "visuele metafoor" van de volle- vs stippellijnen & grijs vs wit is "gesloten" vs "open", maar we zetten mogelijks beide op één lijn wat verwarrend kan zijn. misschien dat Status → Vergadering Status moet worden (of misschien is dit enkel verwarrend voor mij :P)

sergiofenoll committed:
Sort agendas fix & redirect from / to /overzicht

sergiofenoll committed:
Remove unneeded change

sergiofenoll committed:
Undo changes to tests

sergiofenoll committed:
Remove unneeded changes

ValenberghsSven commented:
> Ter info, nu ziet de lijst er (mogelijks) zo uit ![image](https://user-images.githubusercontent.com/22411874/164228126-a3ecd443-5c66-4e09-8ed5-8f42fef80cde.png) Dit is gegeven dat er een vergadering niet afgesloten is waarvan de ontwerpagenda verwijderd is. Dan komt in de lijst enkel de "finale" agenda terecht (dus zonder stippelijn) met ernaast als status "Geopend".
> 
> ik vermeld het omdat in mijn ogen de "visuele metafoor" van de volle- vs stippellijnen & grijs vs wit is "gesloten" vs "open", maar we zetten mogelijks beide op één lijn wat verwarrend kan zijn. misschien dat Status → Vergadering Status moet worden (of misschien is dit enkel verwarrend voor mij :P)

Er is een verschil tussen een niet afgesloten vergadering met enkel goedgekeurde agendas en een afgesloten vergadering.
Het eerste gaat zelden voorvallen, en als het gebeurt is de view uit jouw screenshot ok. Vaak volgt er kort na een afsluiting of een nieuwe ontwerpagenda.
Bij de kanselarij gebruiken ze meestal de term "agenda" om een vergadering te benoemen, " Vergadering Status" is niet gewenst in de views. Dat maakt het soms verwarrend omdat wij denken in termen van data modellen.

erikap commented on app/pods/agendas/template.hbs:
```diff
@@ -19,12 +40,176 @@
+              {{#each @model as |agenda|}}
+                <tr
+                  class="auk-u-text-align--center auk-table--interaction-cursor auk-table--clickable-rows"
+                  {{! template-lint-disable no-invalid-interactive }}
```
You can resolve this by adding `role=button` on the `<tr>`-element

erikap commented on app/pods/agendas/template.hbs:
```diff
@@ -19,12 +40,176 @@
+                      {{/if}}
+                      <span>–</span>
+                      <span>
+                        {{moment-format agenda.createdFor.plannedStart "DD.MM.YYYY"}}
```
Date format should not longer be provided, so we fallback to the configured default (as soon as #1295 has been merged)

sergiofenoll replied:
set it to dashes just so it fits with the other column, will remove it once #1295 gets merged

erikap commented on app/pods/agendas/template.hbs:
```diff
@@ -19,12 +40,176 @@
+                    {{else}}
+                      {{moment-format
+                        agenda.modified
+                        "DD.MM.YYYY — HH:mm"
```
Date format has been decided to be `DD-MM-YYYY ...` (usertest is not yet in line with the default that has been decided recently)

erikap commented on app/pods/agendas/controller.js:
```diff
@@ -2,15 +2,45 @@ import Controller from '@ember/controller';
+export default class AgendasController extends Controller {
+  queryParams = ['page', 'size', 'sort', 'filter'];
+
+  @service router;
```
Duplicate injection of `router` service

erikap commented on app/pods/agendas/controller.js:
```diff
@@ -2,15 +2,45 @@ import Controller from '@ember/controller';
+    } else if (date === '') {
+      this.filter = null;
+    }
+    this.page = 0;
```
You only want to reset the page to 0 if `this.filter` actually changed (so in both if-branches). In the current implementation, if you're not on the first page and you start typing slowly in the search field, page is reset to 0 (and filter cleared) before you finish entering the date.

sergiofenoll committed:
Redirect to /overzicht properly

erikap commented on app/pods/agendas/template.hbs:
```diff
@@ -1,16 +1,37 @@
+          data-test-route-agendas-overview-filter-input
+          @block={{true}}
+          @icon="search"
+          placeholder="bv. 01/2021"
```
There is a translation string for this placeholder `date-example`

erikap commented on app/pods/agendas/template.hbs:
```diff
@@ -19,12 +40,176 @@
+                    <span class="auk-o-flex auk-o-flex--vertical-center auk-o-flex--spaced">
+                      {{#if agenda.status.isDesignAgenda}}
+                        <strong>{{t "design-agenda"}}</strong>
+                        <Auk::Pill @skin="active-future">{{agenda.name}}</Auk::Pill>
```
Misschien beter de non-aliased property `agenda.serialnumber` gebruiken. We moeten er rekening mee houden dat legacy agenda's geen serialnumber hebben, en we dan dus geen pill moeten tonen. 

In de collapsed view, waar enkel serialnumber getoond wordt, is er dan niets zichtbaar. Maar dat lijkt op productie nu ook het geval te zijn :thinking:  Misschien even checken met design wat we dan best tonen.

sergiofenoll replied:
heb het alvast aangepast naar serialnumber en met een if gewrapt, maar laat dit dan unresolved tot we gecheckt hebben met de design fellas

ValenberghsSven replied:
Aangezien we nu al niets tonen in de collapsed view op prod zou ik deze niet laten wachten.

erikap commented:
@brenner-company  Mogen de toevoegingen aan [app/styles/auk/_auk-additions.scs](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1288/files#diff-f3147dd3e568115a92db90417676b86db543edbb6e89ae27d823d939f6421aeb) mee gemerged worden of wil je eerst eens naar kijken? Het zijn er wel redelijk wat :see_no_evil:

sergiofenoll committed:
Use serialnumber, reset page correctly, use role="button"

brenner-company commented:
@erikap Die mogen op zich zo mee gemerged worden (zie commenteer binnen https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1288/files#r854525812), maar het zal inderdaad de bedoeling zijn dat ik daar nog eens volledig door ga 😅

sergiofenoll commented:
These three tests fail with the following error after trying to click on the "Details" tab:

> You didn't provide enough string/numeric parameters to satisfy all of the dynamic segments for route agenda.agendaitems.agendaitem. Missing params: agendaitem_id

My guess when first encountering this was that it's related to how we [create the models for the details route](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-2667-new-agenda-overview/app/components/agenda/agenda-header/agenda-tabs.js#L33). Note that we are passing `this.firstAgendaitem` instead of its id.

![image](https://user-images.githubusercontent.com/22411874/164388280-231c2e05-6983-41a6-8c97-aff457566899.png)
![image](https://user-images.githubusercontent.com/22411874/164388309-09482c7e-d768-4e10-a550-fc6a7bf24538.png)
![image](https://user-images.githubusercontent.com/22411874/164388333-21adc18d-2ec7-488a-bc67-08c3fb741852.png)

This one fails with the following error after deleting an agenda while being inside the "Details" tab:

> While generating link to route "agenda.agendaitems.agendaitem.index": Cannot read properties of undefined (reading 'shouldSupersede')

Said route is used in the "Dossier" tab of the agendaitem navbar. Looking up said error online leads to errors related to passing too little models to a `<LinkTo>`. When I started looking into this, my assumption was that the error gets caused because [we don't pass any models to the tabs in the agendaitem nav](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-2667-new-agenda-overview/app/components/agenda/agendaitem/agendaitem-nav.hbs#L2). I assume that we instead implicitly rely on these links picking up the model from the current route. But when we delete an agenda, we're also transitioning away to a different route, so I imagine that might break things.

![image](https://user-images.githubusercontent.com/22411874/164388322-a1a5d186-fbb2-4823-b3bb-ffd104c7036d.png)

The session-kind failed test is because I forgot to update something in the cypress code, and the failing search test I have no idea about.

All my assumptions about the failing tests aside, I don't know why they fail on this branch, but not on dev/other branches. Given that the errors are rather general and, as far as I can tell, unrelated to the changes of this branch.

sergiofenoll committed:
Remove unused components & fix cypress date test

ValenberghsSven commented:
some search.spec tests are dependent on earlier tests (to avoid having to setup the data again), mandatee-assigning.spec is one of them

ValenberghsSven commented on app/models/meeting.js:
```diff
@@ -75,15 +75,6 @@ export default Model.extend({
     return this.isFinal && !this.releasedDocuments;
   }),
 
-  latestAgenda: computed('agendas.[]', function() {
-    return PromiseObject.create({
-      promise: this.get('agendas').then((agendas) => {
-        const sortedAgendas = agendas.sortBy('agendaName').reverse();
-        return sortedAgendas.get('firstObject');
-      }),
-    });
-  }),
-
```
I think you removed `latestAgenda` because of my review?
However, I said `latestAgendaǸame` can go.
`latestAgenda` computed still has more usages (agenda-activity model, pods/agenda/overview, pods/newsletter)

ValenberghsSven replied:
refactor of those 3 places is out of scope so we should just revive the computed

sergiofenoll replied:
Potverdorie, I blame lack of coffee for this... Thanks for the catch Sven!

sergiofenoll committed:
Re-add latestAgenda and remove latestAgendaName

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2667-new-agenda-overview

ValenberghsSven committed:
KAS-2667 calculating firstagendaitem/deleting last agendaitem fixes

ValenberghsSven committed:
KAS-2667 formally ok command better awaits, deleted meeting shouldn't rerender

ValenberghsSven committed:
KAS-2667 move spec files because of bug in yggdrasil

sergiofenoll committed:
Get currentAgendaitem from anchor

When we're rendering agenda tabs from agenda.agendaitems we should try
and determine the currentAgendaitem based on the queryParam ancho, if available.

sergiofenoll committed:
Remove default moment format
